app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -24,6 +24,25 @@ def host_setup_extension
 
         super
       end
+
+      def context_urls
+        urls = super
+        urls[:rhsm_base_url] = rhsm_base_url
+        urls
+      end
+
+      private
+
+      def rhsm_base_url
+        smart_proxy = if params[:url]
+          SmartProxy.find_by(url: params[:url])
```

Comment:
If we are being technically correct, we should ensure that the found smart proxy has the Pulp 3 feature and throw an error if not.